### PR TITLE
Add GPIO interface for imxrt

### PR DESCRIPTION
**Pull request by HarrisonChoi750405**
add gpio interface
 - interrupt handling need to be added in future
 - the api about pull setting need to be added in future

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [45f10c7b521636f9cec38bc421fa8a830db1c2e8] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [45f10c7b521636f9cec38bc421fa8a830db1c2e8] -

File: os/arch/arm/src/imxrt/imxrt_gpio_lowerhalf.c [hunk 1]
```diff
@@ -0,0 +1,260 @@
+/******************************************************************
```
Comment (Taejun-Kwon) [inline os/arch/arm/src/imxrt/imxrt_gpio_lowerhalf.c]: you don't have to add lowerhalf as a postfix.

File: os/arch/arm/src/imxrt/imxrt_gpio_lowerhalf.c [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+ *
+ * Description:
+ *   Instantiate the GPIO lower half driver for the S5J.
+ *   General usage:
+ *
+ *     #include <tinyara/gpio.h>
+ *     #include "imxrt_gpio.h"
+ *
+ *     struct gpio_lowerhalf_s *lower;
+ *     lower = 5j_gpio_lowerhalf();
```
Comment (Taejun-Kwon) [inline os/arch/arm/src/imxrt/imxrt_gpio_lowerhalf.c]: Could you please explain what 5j means?

**Comment by seinfra** [comment]
Target : [83d99771d432f39a5425489e100dfcf37a108083] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [83d99771d432f39a5425489e100dfcf37a108083] - Code Rule Check OK.

**Pull request merged**